Sources/StreamChat/Repositories/AuthenticationRepository.swift

```diff
@@ -295,12 +285,17 @@ class AuthenticationRepository {
 
     private func scheduleTokenFetch(isRetry: Bool, userInfo: UserInfo?, tokenProvider: @escaping TokenProvider, completion: @escaping (Error?) -> Void) {
         guard !isGettingToken || isRetry else {
-            tokenRequestCompletions.append(completion)
+            tokenQueue.async(flags: .barrier) {
+                self._tokenRequestCompletions.append(completion)
+            }
             return
         }
 
+        let interval = tokenQueue.sync(flags: .barrier) {
+            _tokenExpirationRetryStrategy.getDelayAfterTheFailure()
+        }
```

Comment:
`getDelayAfterTheFailure` is mutating function, therefore we must use `.barrier`